driver-sync/src/main/com/mongodb/client/internal/UuidRepresentationOverridingCodecRegistry.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.internal;
```

Comment:
In order to use this in the async drivers, this will have to move to `org.bson.internal`.  Or should it be public?  If the former, then CycleDetectingCodecRegistry and ChildCodecRegistry could also be internal

Reply:
Moved in second patch